bazel/internal/scan/scan.bzl

```diff
@@ -165,8 +165,12 @@ def scan_impl(ctx):
     xcmappingmodels = sets.to_list(xcmappingmodels_set)
     test_targets = sets.to_list(test_targets_set)
 
+    indexstores_config = [file.path for file in indexstores]
+    if ctx.attr.global_indexstore:
+        indexstores_config = [ctx.attr.global_indexstore]
+
     project_config = struct(
-        indexstores = [file.path for file in indexstores],
+        indexstores = indexstores_config,
```

Comment:
I created this separately instead of modifying the `indexstores` value so that the individual files will be included in the runfiles of the rule and Bazel will consider them as dependencies. Not 100% sure if that's actually required or not though